sample/src/main/java/com/airwallex/paymentacceptance/repo/LocalMockRepository.kt

```diff
@@ -0,0 +1,107 @@
+        val response = api.authentication(Settings.apiKey, Settings.clientId)
+        Settings.token = JSONObject(response.string())["token"].toString()
+    }
+
+    override suspend fun getPaymentIntentFromServer(
+        force3DS: Boolean,
+        customerId: String?
+    ): PaymentIntent {
+        if (customerId == null) {
+            login()
+        }
+        return withContext(Dispatchers.IO) {
+            val body = mutableMapOf(
+                "request_id" to UUID.randomUUID().toString(),
+                "amount" to Settings.price.toDouble(),
+                "currency" to Settings.currency,
+                "merchant_order_id" to UUID.randomUUID().toString(),
+                "order" to PurchaseOrder.Builder()
+                    .setProducts(products)
+                    .setShipping(shipping)
+                    .setType("physical_goods")
+                    .build()
+                    .toParamMap(),
+                "descriptor" to "Airwallex - T-sh  irt",
+                "metadata" to mapOf("id" to 1),
+                "email" to "yimadangxian@airwallex.com",
+                "return_url" to Settings.returnUrl
+            )
+            if (force3DS) {
+                body["payment_method_options"] =
+                    mapOf("card" to mapOf("three_ds_action" to "FORCE_3DS"))
+            }
+            Settings.cachedCustomerId?.let { body.put("customer_id", it) }
+            customerId?.let { body.put("customer_id", it) }
+            val paymentIntentResponse = api.createPaymentIntent(body)
+            PaymentIntentParser().parse(JSONObject(paymentIntentResponse.string()))
+        }
+    }
+
+    override suspend fun getCustomerIdFromServer(): String {
+        return withContext(Dispatchers.IO) {
+            login()
+            if (Settings.cachedCustomerId.isNullOrEmpty()) {
+                val customerResponse = api.createCustomer(
+                    mutableMapOf(
+                        "request_id" to UUID.randomUUID().toString(),
+                        "merchant_customer_id" to UUID.randomUUID().toString(),
+                        "first_name" to "John",
+                        "last_name" to "Doe",
+                        "email" to "john.doe@airwallex.com",
+                        "phone_number" to "13800000000",
+                        "additional_info" to mapOf(
+                            "registered_via_social_media" to false,
+                            "registration_date" to "2019-09-18",
+                            "first_successful_order_date" to "2019-09-18"
+                        ),
+                        "metadata" to mapOf(
+                            "id" to 1
+                        )
+                    )
+                )
+                val customerId = JSONObject(customerResponse.string())["id"].toString()
+                Settings.cachedCustomerId = customerId
+                customerId
+            } else {
+                Settings.cachedCustomerId!!
```

Comment:
This force unwrapping can be avoided by:
`
val cachedCustomerId = Settings.cachedCustomerId
if (cachedCustomerId.isNullOrEmpty()) {
} else {
     cachedCustomerId
}
`

Reply:
ok